### ISSUE TITLE
Add ability to set k8s service type in pipeline spec 

### ISSUE THREAD

**Issue by Nick-Harvey** [pachyderm/pachyderm#3650]
**What is the goal / desired outcome?**
When creating pipelines that have "service" set in the pipeline spec, it would be nice to have the option to declare the service-type as well so that the user doesn't have to manually edit the service to change it. 

I understand our current functionality is "as designed" for security reasons, however, for quick iteration in environments with fewer security requirements this would save the user some work (and kubernetes work at that ;) ) 

**If there is a way to accomplish this today via workaround, what does that require?**
Right now users just have to edit the service manually. 

**(Optional) What is your proposal for a feature to solve this?**
in the pachyderm pipeline spec, specifically the service section, a type field. So something like
```
"service": {
    "internal_port": int,
    "external_port": int,
    "type": "LoadBalancer" or "ExternalName" or "NodePort" (current default behavior) 
  },
```

**Environment?**:
- Kubernetes version (use `kubectl version`):
- Pachyderm CLI version (use `pachctl version`):
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
- OS (e.g. from /etc/os-release):
- Others:

**Comment by DanielMorales9** [comment]
Hi, how much effort do you see in implementing this? What would be the affected components?

**Comment by Nick-Harvey** [comment]
I'm not one of the core maintainers so it would be hard for me to say, but I don't think it would require too much effort.

**Comment by ysimonson** [comment]
Related: #3137

**Comment by gabrielgrant** [comment]
I think this is fully addressed by #3753 (thanks @DanielMorales9 !)
